charts/consul/templates/_helpers.tpl

```diff
@@ -22,6 +22,13 @@ as well as the global.name setting.
             {{ "{{" }}- end -{{ "}}" }}
 {{- end -}}
 
+{{- define "consul.vaultDecodedSecretTemplate" -}}
+ |
+            {{ "{{" }}- with secret "{{ .secretName }}" -{{ "}}" }}
+            {{ "{{" }}- {{ printf "base64Decode .Data.data.%s" .secretKey }} -{{ "}}" }}
+            {{ "{{" }}- end -{{ "}}" }}
+{{- end -}}
```

Comment:
Why do we need to base64 decode it? Does vault kv engine not work with not-encoded config? Or does it encode it for you if similar to k8s secrets?

Reply:
Great question!  Let me go back and check that I did not encode as part of the test set up.

Reply:
I totally was.  I used json.Marshall() and then saving that to Vault instead of wrapping that result in string() and then saving it.  I have a PR with the acceptance test and made the modification there ...since I can test it.  This is the commit: https://github.com/hashicorp/consul-k8s/pull/1125/commits/634164a3b2db74148249293ceb78238187ab6a13